content/docs/pipelines/reference/component-spec.md

```diff
@@ -0,0 +1,137 @@
+    * `type`: Specifies the type of input/output. The types are used
+        as hints for pipeline authors and can be used by the pipeline system/UI
+        to validate arguments and connections between components. Basic types
+        are **String**, **Integer**, **Float**, and **Bool**. See the full list
+        of [types](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/dsl/types.py)
+        defined by the Kubeflow Pipelines SDK.
+
+### Implementation
+
+* `implementation`: Specifies how to execute the component instance.
+    There are two implementation types,  `container` and `graph`. (The latter is
+    not in scope for this document.) In future we may introduce more 
+    implementation types like `daemon` or `K8sResource`.
+
+    * `container`:
+        Describes the Docker container that implements the component. A portable 
+        subset of the Kubernetes
+        [Container v1 spec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.12/#container-v1-core).
+
+        * `image`: Name of the Docker image.
+        * `command`: Entrypoint array. The Docker image's
+            ENTRYPOINT is used if this is not provided. Each item is either a
+            string or a placeholder. The most common placeholders are
+            `{inputValue: Input name}` and `{outputPath: Output name}`.
+        * `args`: Arguments to the entrypoint. The Docker
+            image's CMD is used if this is not provided. Each item is either a
+            string or a placeholder. The most common placeholders are
+            `{inputValue: Input name}` and `{outputPath: Output name}`.
+        * `env`: Map of environment variables to set in the container.
+        * `fileOutputs`: Legacy property that is only needed in
+            cases where the container always stores the output data in some
+            hard-coded non-configurable local location. This property specifies
+            a map between some outputs and local file paths where the program
+            writes the output data files. Only needed for components that have
+            hard-coded output paths. Such containers need to be fixed by
+            modifying the program or adding a wrapper script that copies the
+            output to a configurable location. Otherwise the component may be
+            incompatible with future storage systems.
+
+You can set all other Kubernetes container properties when you
+use the component inside a pipeline.
+
+## Using placeholders for command-line arguments
+
+### Consuming input by value (parameter)
+
+The placeholder is replaced by the contents of the input argument.
+
+* In `component.yaml`: `{inputValue: Rounds}`
+* Resulting Argo YAML: `"{{input.parameters.rounds}}"`
+
+### Outputs
+
+Output paths are filled in by the pipeline system. The Output construct is
+replaced by a path. (The path can possibly point to a mounted output volume.)
+
+* In `component.yaml`: `{outputPath: Trained model}`
```

Comment:
In `component.yaml`:
```command: [program.py, --out-model, {outputPath: trained_model}]```
In pipeline code:
```task1 = component1()```
Resulting command-line (the placeholder is substituted with generated path):
```program.py --out-model /outputs/trained_model/data```

Reply:
Done.